packages/input-datepicker/src/LionInputDatepicker.js

```diff
@@ -213,12 +216,19 @@ export class LionInputDatepicker extends LionInputDate {
     }
   }
 
+  __onReadOnlyChanged() {
+    if (this._invokerElement) {
+      this._invokerElement.disabled = this.disabled || this.readOnly;
+    }
+  }
+
   /**
    * TODO: [delegation of disabled] move this to LionField (or FormControl) level
    */
   firstUpdated(c) {
     super.firstUpdated(c);
     this.__delegateDisabled();
+    this.__onReadOnlyChanged();
```

Comment:
`__onReadOnlyChanged` is not the best name when you need to call it from multiple place like here in `firstUpdated`, and not only from `_requestUpdate`

I would rather give it a name that suits both needs, and I think the current name for `disabled` is way better in that regard, but my comment above shows that you don't really delegate it atm... I think you should delegate as well (explained above) and then the name will be `__delegateReadOnly`, so this can be resolved too

Reply:
Readonly is not delegated, but used as a syncObserver (we need to consider both disabled and readonly instead of just putting it blindly on invokerElement). It's a common pattern throughout our code to handle all logic that needs to be executed in an observer '__onPropXChanged'

Reply:
I changed it to `__toggleDisabledOfInvokerOnReadonlyChanged`. Please change and push if you don't agree ;)

Reply:
does not explain why you don't delegate 'readonly' to the input element? I didn't ask to set such an attribute on the invoker, but rather delegate and set it on the input element

let's agree on the name after we agree on the logic here